packages/headless/src/utils/compare-utils.ts

```diff
@@ -0,0 +1,27 @@
+import {createCustomEqual} from 'fast-equals';
+
+export function arrayEqualSorted<T>(firstArray: T[], secondArray: T[]) {
+  return (
+    firstArray.length === secondArray.length &&
+    firstArray.findIndex((val, i) => secondArray[i] !== val) === -1
+  );
+}
+
+export const deepObjectEqualUnsorted = createCustomEqual(
```

Comment:
The `unsorted` keywords is a bit strange. An object is not really "sortable". 

I understand it's there for "nested" array values but... 

I'd go with just `deepEqual` ? 🤔

Reply:
I agree. I think `deepEqual` is a suitable name that explains what the function does.

Reply:
The deepEqual is the original, and for it, the order of array values matter, that's why I put the unsorted, I could put disordered, not sure...

Reply:
Called it `deepEqualAnyOrder`